### PR TITLE
Upgrade to Mapbox Maps SDK for iOS v6.0.0

### PR DESCRIPTION
**Pull request by julianrex**
This PR updates the Maps SDK to ~~5.10.0-beta.1~~ v6.0.0 and fixes CI to reflect new SDK registry workflow.

### CODE REVIEW

**Comment by 1ec5** [comment]
This PR needs to be rebased onto release-v1.0-pre-registry.

**Comment by nishant-karajgikar** [comment]
@1ec5 , Just rebased on release-v1.0-pre-registry and updated maps dependency to 5.10.0-beta.1

File: README.md [hunk 1]
```diff
@@ -34,6 +34,15 @@ The Mapbox Navigation SDK is also available [for Android](https://github.com/map
...
 
+1. Create a Mapbox API token (from your account page on [mapbox.com](https://www.mapbox.com)) with the      `DOWNLOADS:READ` scope. 
+   **PLEASE NOTE: This is not the same as your production Mapbox API token.** 
+   Once you have the token, create a `~/.netrc` which looks like this:
+   ```
+   machine api.mapbox.com 
+      login mapbox
+      password <INSERT API TOKEN>
+   ```
+
```
Comment (1ec5) [inline README.md:45]: This is the section about installing the latest stable release, v0.40.0. Can you move these three changes down to the section about installing the beta?

We’ll remove this section once we publish a final release of v1.0.0. In the meantime, by the time map SDK v5.10.0 goes out, we’ll have to publish a v0.40.1 that prevents CocoaPods and Carthage from automatically upgrading to v5.10.0: #2447.

File: README.md [hunk 2]
```diff
@@ -50,7 +50,51 @@ Alternatively, to install Mapbox Navigation using [Carthage](https://github.com/
    github "mapbox/mapbox-navigation-ios" ~> 0.40
    ```
 
-1. Run `carthage update --platform iOS` to build just the iOS dependencies.
+1. Follow the rest of [Carthage’s iOS integration instructions](https://github.com/Carthage/Carthage#if-youre-building-for-ios-tvos-or-watchos). Your application target’s Embedded Frameworks should include `MapboxNavigation.framework`, `MapboxCoreNavigation.framework`, `MapboxNavigationNative.framework`, and `MapboxAccounts.framework`.
+
+## Installing the latest stable version with the Maps SDK 5.10.0-beta version
```
Comment (1ec5) [inline README.md:55]: There’s already a section about this below, called “Installing the latest prerelease”.

Comment (dersim-davaod) [reply README.md]: you're right. the latest version of the branch contains it after rebasing. updated.

**Comment by 1ec5** [comment]
Got a failure in [the Xcode_11.4.1_iOS_12.2 build](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/1257/workflows/610f3c14-f811-4c13-bf90-64125e17921a/jobs/17420/steps), possibly intermittent:

```
Test Case '-[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirectionsError]' started.

2020-07-17 16:59:19.439880+0000 Example[68911:118130] [Common] _BSMachError: port 8b03; (os/kern) invalid capability (0x14) "Unable to insert COPY_SEND"

2020-07-17 16:59:19.443302+0000 Example[68911:118130] [Common] _BSMachError: port 8b03; (os/kern) invalid capability (0x14) "Unable to insert COPY_SEND"

<unknown>:0: error: -[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirectionsError] : Asynchronous wait failed: Exceeded timeout of 2 seconds, with unfulfilled expectations: "Set up offline routing".

Test Case '-[MapboxCoreNavigationTests.OfflineRoutingTests testOfflineDirectionsError]' failed (3.089 seconds).

Test Case '-[MapboxCoreNavigationTests.OfflineRoutingTests testUnpackTilePack]' started.

2020/07/17 16:59:22.109889 [ERROR] Invalid tile path: foo/baz

2020/07/17 16:59:22.127649 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.127924 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128133 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128284 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128389 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128498 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128610 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128689 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.128876 [ERROR] Invalid tile path: 

2020/07/17 16:59:22.129131 [ERROR] Invalid tile path: 

Test Case '-[MapboxCoreNavigationTests.OfflineRoutingTests testUnpackTilePack]' passed (0.965 seconds).
```

**Pull request merged**